### PR TITLE
SUBMARINE-490. Synchronize user information from LDAP

### PR DESCRIPTION
**Pull request by Eroschang**
### What is this PR for?
Test to synchronize user information from LDAP server.

### What type of PR is it?
Feature

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/SUBMARINE/
* Put link here, and add [SUBMARINE-*Jira number*] in PR title, eg. [SUBMARINE-23]
https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-490?filter=allopenissues&orderby=priority+ASC%2C+updated+DESC

### How should this be tested?
* First time? Setup Travis CI as described on https://submarine.apache.org/contribution/contributions.html#continuous-integration
* Strongly recommended: add automated unit tests for any new or changed behavior
* Outline any manual steps to test the PR here.

### Screenshots (if appropriate)
Edit the example.ldif file to add user in LDAP server.
image

Use testList() to test and get the user name which in LDAP server.
image

The user name in LDAP server.
image

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-commons/commons-unixusersync/src/test/resources/example.ldif [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+objectClass: domain
+objectClass: top
+dc: zapodot
+
+dn: ou=groups,dc=zapodot,dc=org
+objectclass: top
+objectclass: organizationalUnit
+ou: groups
+
+
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/resources/example.ldif:31]: Keep only one blank line

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import javax.naming.directory.DirContext;
+import javax.naming.directory.SearchControls;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
+public class EmbeddedLdapRuleTest {
+
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:48]: Delete this blank line

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+import javax.naming.NameClassPair;
+import javax.naming.NamingEnumeration;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.SearchControls;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:46]: Keep only one blank line

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            ldapConnection.search(DOMAIN_DSN, SearchScope.SUB, "(objectClass=person)");
+
+    assertEquals(2, searchResult.getEntryCount());
+  }
+
+  @Test
+  public void testRawLdapConnection() throws Exception {
+    final String commonName = "Test person";
+    final String dn = String.format(
+            "cn=%s,ou=people,dc=zapodot,dc=org", commonName);
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:70]: No line breaks

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    LDAPConnection ldapConnection = embeddedLdapRule.unsharedLdapConnection();
+    try {
+      ldapConnection.add(new AddRequest(dn, Arrays.asList(
+              new Attribute("objectclass", "top",
+                      "person", "organizationalPerson", "inetOrgPerson"),
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:75]: No line breaks

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    public void testContextClose() throws Exception {
+    final Context context = embeddedLdapRule.context();
+    context.close();
+    assertNotNull(context.getNameInNamespace());
+  }
+
+  @Test
+    public void testEmbeddedServerPort() throws Exception {
+    assertTrue(embeddedLdapRule.embeddedServerPort() > 0);
+
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:130]: Delete blank line

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  }
+
+  private void assertTrue(boolean b) {
+  }
+
+  @Test(expected = IllegalStateException.class)
+    public void testNoPortAssignedYet() throws Exception {
+    final EmbeddedLdapRule embeddedLdapRule = new EmbeddedLdapRuleBuilder().build();
+    embeddedLdapRule.embeddedServerPort();
+
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:140]: Delete blank line

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    assertNotNull(context.getNameInNamespace());
+  }
+
+  @Test
+    public void testEmbeddedServerPort() throws Exception {
+    assertTrue(embeddedLdapRule.embeddedServerPort() > 0);
+
+  }
+
+  private void assertTrue(boolean b) {
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:133]: Delete this empty function.

File: submarine-commons/commons-unixusersync/src/test/resources/example.ldif [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+objectclass: top
+objectclass: organizationalUnit
+ou: people
+
+dn: ou=group-2,dc=zapodot,dc=org
+objectclass: top
+objectclass: organizationalUnit
+ou: people
+
+
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/resources/example.ldif:48]: Keep only one blank line

File: submarine-commons/commons-unixusersync/pom.xml [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    <dependency>
+      <groupId>org.zapodot</groupId>
+      <artifactId>embedded-ldap-junit</artifactId>
+      <version>0.7</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.zapodot</groupId>
+      <artifactId>embedded-ldap-junit</artifactId>
+      <version>0.7</version>
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/pom.xml:53]: define `embedded-ldap-junit.version=0.7` in `submarine/pom.xml`

File: submarine-commons/commons-unixusersync/pom.xml [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+      <groupId>commons-configuration</groupId>
+      <artifactId>commons-configuration</artifactId>
+      <version>${commons-configuration.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.zapodot</groupId>
+      <artifactId>embedded-ldap-junit</artifactId>
+      <version>0.7</version>
+    </dependency>
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/pom.xml:48]: Delete this duplicate import

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java (hunk 5 above)
Comment (Eroschang) [reply submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:75]: Well... there will more than 110 words without this line breaks.

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java (hunk 8 above)
Comment (Eroschang) [reply submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:133]: This function is for the testEmbeddedServerPort().

File: submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  @Test
+  public void testList() throws Exception {
+    final Context context = embeddedLdapRule.context();
+    NamingEnumeration list = context.list("ou=semi-people,dc=zapodot,dc=org");
+
+    while (list.hasMore()){
+      NameClassPair nc = (NameClassPair) list.next();
+      System.out.println(nc.getName());
```
Comment (liuxunorg) [inline submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java:112]: Use LOG.info(xxxx)

```
private static final Logger LOG = LoggerFactory.getLogger(EmbeddedLdapRuleTest.class);
```

**Pull request closed without merging**